backend/src/main/java/io/spring/cloud/samples/animalrescue/backend/SecurityConfiguration.java

```diff
@@ -1,51 +1,46 @@
 package io.spring.cloud.samples.animalrescue.backend;
 
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import io.pivotal.cfenv.core.CfEnv;
-
+import org.springframework.boot.autoconfigure.condition.ConditionalOnCloudPlatform;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 import org.springframework.context.annotation.Profile;
 import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.core.userdetails.MapReactiveUserDetailsService;
+import org.springframework.security.core.userdetails.ReactiveUserDetailsService;
+import org.springframework.security.core.userdetails.User;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.server.SecurityWebFilterChain;
+import org.springframework.security.web.server.authentication.RedirectServerAuthenticationSuccessHandler;
 
 @Configuration
-@Profile("!local")
+@Profile("!cloud") // cloud profile is automatically activated on CloudFoundry
 public class SecurityConfiguration {
 
-	private static final Logger LOG = LoggerFactory.getLogger(SecurityConfiguration.class);
-
 	@Bean
-	CfEnv cfEnv() {
-		return new CfEnv();
+	PasswordEncoder passwordEncoder() {
+		return new BCryptPasswordEncoder();
 	}
 
 	@Bean
-	public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity httpSecurity, CfEnv cfEnv) {
-		if (cfEnv.isInCf()) {
-			String authDomain = cfEnv.findCredentialsByLabel("p.gateway").getString("auth_domain");
-			if (authDomain != null) {
-				LOG.info("not in cf");
-				httpSecurity.oauth2ResourceServer()
-					.jwt().jwkSetUri(authDomain + "/token_keys");
-			}
-		}
+	ReactiveUserDetailsService reactiveUserDetailsService(PasswordEncoder passwordEncoder) {
+		return new MapReactiveUserDetailsService(
+			User.withUsername("alice").password(passwordEncoder.encode("test")).roles("USER").build(),
+			User.withUsername("bob").password(passwordEncoder.encode("test")).roles("USER").build()
+		);
+	}
 
+	@Bean
+	public SecurityWebFilterChain securityWebFilterChain(ServerHttpSecurity httpSecurity) {
 		// @formatter:off
 		return httpSecurity
 			.httpBasic().disable()
+			.formLogin().authenticationSuccessHandler(new RedirectServerAuthenticationSuccessHandler("http://localhost:3000/rescue")).and()
 			.csrf().disable()
```

Comment:
I noticed that we are not setting any of these security rules in `CloudFoundrySecurityConfiguration`. Have you tried deploying the app to CF and validate that it works? We don't have any test around it, so I wanted to double check with you.

Reply:
Yep, that works just fine in CF. When Gateway handles OAuth2 flow, it sends a user back to `/rescue/login` which then forwards a user to `/rescue`. 

We can tryt to refactor e2e sutie to make it work with CF + gateway deployment.